components/state_machine/__init__.py

```diff
@@ -129,32 +228,72 @@ def output_graph(config):
 
     return config
 
+
+MAX_CONDITION_LENGTH_IN_DIAGRAM = 22
 def output_mermaid_graph(config):
     graph_data = f"stateDiagram-v2{os.linesep}"
     graph_data = graph_data + f"  direction LR{os.linesep}"
     initial_state = config[CONF_INITIAL_STATE] if CONF_INITIAL_STATE in config else config[CONF_STATES_KEY][0][CONF_NAME]
     graph_data = graph_data + f"  [*] --> {initial_state}{os.linesep}"
+    footnotes = []
+
     for input in config[CONF_INPUTS_KEY]:
         if CONF_INPUT_TRANSITIONS_KEY in input:
             for transition in input[CONF_INPUT_TRANSITIONS_KEY]:
-                graph_data = graph_data + f"  {transition[CONF_FROM]} --> {transition[CONF_TO]}: {input[CONF_NAME]}{os.linesep}"
+                # TODO: significant duplicated code with the DOT graph?
```

Comment:
I'm thinking to removing the DOT graph support and only using Mermaid format. This would reduce code duplication and effort to support this